src/main/scala/org/ergoplatform/nodeView/wallet/ErgoAddress.scala

```diff
@@ -0,0 +1,213 @@
+package org.ergoplatform.nodeView.wallet
+
+import java.nio.ByteBuffer
+import java.util
+
+import com.google.common.primitives.Ints
+import org.ergoplatform.settings.{Algos, ErgoSettings}
+import scapi.sigma.DLogProtocol.ProveDlog
+import scorex.crypto.encode.Base58
+import scorex.crypto.hash.Blake2b256
+import sigmastate._
+import sigmastate.Values.{IntConstant, TaggedByteArray, Value}
+import sigmastate.serialization.{DataSerializer, ValueSerializer}
+import sigmastate.utils.ByteBufferReader
+import sigmastate.utxo.{DeserializeContext, Slice}
+
+import scala.util.Try
+
+
+/**
+  * An address is a short string corresponding to some script used to protect a box. Unlike (string-encoded) binary
+  * representation of a script, an address has some useful characteristics:
+  *
+  * * Integrity of an address could be checked., as it is incorporating a checksum.
+  * * A prefix of address is showing network and an address type.
+  * * An address is using an encoding (namely, Base58) which is avoiding similarly l0Oking characters, friendly to
```

Comment:
do we really use base58 still?

also are the double ** stars – something intended?